x/farming/client/testutil/suite.go

```diff
@@ -601,3 +599,498 @@ func (s *IntegrationTestSuite) TestNewHarvestCmd() {
 		})
 	}
 }
+
+type QueryCmdTestSuite struct {
+	suite.Suite
+
+	cfg     network.Config
+	network *network.Network
+}
+
+func (s *QueryCmdTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	db := tmdb.NewMemDB()
+	cfg := NewConfig(db)
+	cfg.NumValidators = 1
+
+	var genesisState farmingtypes.GenesisState
+	err := cfg.Codec.UnmarshalJSON(cfg.GenesisState[farmingtypes.ModuleName], &genesisState)
+	s.Require().NoError(err)
+
+	genesisState.Params = farmingtypes.DefaultParams()
+	cfg.GenesisState[farmingtypes.ModuleName] = cfg.Codec.MustMarshalJSON(&genesisState)
+	cfg.AccountTokens = sdk.NewInt(100_000_000_000) // node0token denom
+	cfg.StakingTokens = sdk.NewInt(100_000_000_000) // stake denom
+
+	s.cfg = cfg
+	s.network = network.New(s.T(), cfg)
+
+	_, err = s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+
+	s.createFixedAmountPlan(
+		"test",
+		sdk.NewDecCoins(sdk.NewInt64DecCoin(sdk.DefaultBondDenom, 1)),
+		types.ParseTime("0001-01-01T00:00:00Z"),
+		types.ParseTime("9999-01-01T00:00:00Z"),
+		sdk.NewCoins(sdk.NewInt64Coin("node0token", 100_000_000)),
+	)
+	s.fundFarmingPool(1, sdk.NewCoins(sdk.NewInt64Coin("node0token", 1_000_000_000)))
+	s.stake(sdk.NewCoins(sdk.NewInt64Coin(sdk.DefaultBondDenom, 1000000)))
+	s.advanceEpoch()
+	s.advanceEpoch()
+}
+
+func (s *QueryCmdTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
+}
+
+func (s *QueryCmdTestSuite) createFixedAmountPlan(name string, stakingCoinWeights sdk.DecCoins, startTime, endTime time.Time, epochAmount sdk.Coins) {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+
+	req := cli.PrivateFixedPlanRequest{
+		Name:               name,
+		StakingCoinWeights: stakingCoinWeights,
+		StartTime:          startTime,
+		EndTime:            endTime,
+		EpochAmount:        epochAmount,
+	}
+	file := testutil.WriteToNewTempFile(s.T(), req.String())
+	defer os.Remove(file.Name())
+
+	args := append([]string{
+		file.Name(),
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, val.Address.String()),
+	}, commonArgs...)
+
+	cmd := farmingcli.NewCreateFixedAmountPlanCmd()
+
+	_, err := clitestutil.ExecTestCLICmd(clientCtx, cmd, args)
+	s.Require().NoError(err)
+}
+
+func (s *QueryCmdTestSuite) fundFarmingPool(poolId uint64, amount sdk.Coins) {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+	types.RegisterInterfaces(clientCtx.InterfaceRegistry) // TODO: can we remove this line?
```

Comment:
```suggestion
	types.RegisterInterfaces(clientCtx.InterfaceRegistry)
```
It looks like a normal registration case, Let's remove the TODO: comments